controllers/users.js

```diff
@@ -410,6 +410,31 @@ const updateSelf = async (req, res) => {
       }
     }
 
+    if (req.body.disabled_roles) {
+      let rolesToDisable;
+      const data = req.body.disabled_roles;
+
+      if (user.disabled_roles !== undefined) {
+        rolesToDisable = user.disabled_roles;
+
+        data.forEach((role) => {
+          const roleIndex = rolesToDisable.indexOf(role);
+          if (roleIndex !== -1) {
+            rolesToDisable.splice(roleIndex, 1);
+          } else {
+            rolesToDisable.push(role);
+          }
+        });
+      } else {
+        rolesToDisable = data;
+      }
+
+      const updatedUser = await userQuery.addOrUpdate({ disabled_roles: rolesToDisable }, userId);
+      if (updatedUser) {
+        return res.status(200).send({ message: "Privilege modified successfully!" });
+      }
```

Comment:
Also does this API specifically modifies privilege?

Reply:
No, it's meant for other purposes as well, As suggested I tried to reuse the `Patch /users/self` route

Reply:
So how is the role being updated like and user data being updated?
Will the roles be only edited in single call? or can roles be edited with other user data?

Are you not considering disabled_roles as part of user data. 
will you update only disabled_roles ?

Reply:
> So how is the role being updated like and user data being updated?

we have a `query.addOrupdate()` takes in userId and properties to add in our case `{disabled_roles: [] }`

> will you update only disabled_roles ?

Yes, the effect of toggle of the super_user privileges happens somewhere else, in the `fetchUser()` function

> Are you not considering disabled_roles as part of user data.

No, I'm considering disabled_roles as the part of userData. addOrUpdate adds this disable_roles property to the userObject